mobile/src/shared/SharedHooks.ts

```diff
@@ -0,0 +1,20 @@
+import type { Dispatch, SetStateAction } from "react";
+
+import { useOutletContext } from "react-router-dom";
+
+import type { NullableUser, NullableUserStateType } from "./SharedTypes";
+import { RemoteUserRepository, UserRepository } from "../user/UserRepository";
+
+interface NullableUserContextType {
+  user: NullableUser;
+  setUser: Dispatch<SetStateAction<NullableUser>>;
+}
+
+export function useUser(): NullableUserStateType {
+  const { user, setUser } = useOutletContext<NullableUserContextType>();
```

Comment:
The object to array conversion is probably avoidable if you set the outlet context as an array in the first place. This would remove the need for a custom type above.